### DISCUSSION TITLE
Implementation of ParamListRegisterOut.java maybe incorrect

### DISCUSSION THREAD

**Discussion by Wall-AF** [NationalSecurityAgency/ghidra#4101] [Q&A]
Can someone verify that the implementation of <https://github.com/NationalSecurityAgency/ghidra/blob/8598f28b230d4e706749f1fcb25570c733889447/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/lang/ParamListRegisterOut.java>?

It seems there could be a loop missing.

**Comment by astrelsky** [comment]
I think the loop you are expecting to see is in `assignAddress`. The `status` array gets modified here https://github.com/NationalSecurityAgency/ghidra/blob/8598f28b230d4e706749f1fcb25570c733889447/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/lang/ParamListStandard.java#L99

I have no idea what the code is supposed to do and this was just an observation.

**Reply by Wall-AF** [reply]
Oops, misread the situation!
